### PR TITLE
Player movement based on coordinates

### PR DESCRIPTION
**Pull request by TheoPannetier**
**Test for:**

- [x] #336 

**Fixes:**

- [x] #366
- [x] #336

### CODE REVIEW

File: coordinate.cpp [hunk 1]
```diff
@@ -49,35 +61,41 @@ void test_coordinate() {
 
     double default_speed = 1;
     c.move(direction);
-    assert(get_x(c) == default_speed);
-    assert(get_y(c) == 0 );
+    double error_x = abs(get_x(c) - default_speed);
+    double error_y = abs(get_y(c) - 0);
+    assert(error_x < 0.00001);
+    assert(error_y < 0.00001);
```
Comment (ollyturner) [inline coordinate.cpp:67]: We don't really need this bit. Line 69 covers all this.

File: coordinate.cpp [hunk 2]
```diff
@@ -49,35 +61,41 @@ void test_coordinate() {
...
       }
 
     ///Test along y axis
     ///When direction is 0 player should
     ///only move along the y axis
-    direction = M_PI/2;
-
-    double default_speed = 1;
+    c.reset_coords_to_zero();
+    direction = M_PI / 2;
```
Comment (ollyturner) [inline coordinate.cpp:84]: We should use a vector of directions, like line 69, to test all the cases.

File: coordinate.cpp [hunk 3]
```diff
@@ -49,35 +61,41 @@ void test_coordinate() {
...
     c.move(direction);
-    assert(get_x(c) == 0);
-    assert(get_y(c) == default_speed );
+    error_x = abs(get_x(c) - 0);
+    error_y = abs(get_y(c) - default_speed);
+    assert(error_x < 0.00001);
+    assert(error_y < 0.00001);
 
-    std::vector<double> speeds{0.1,0.5,2,-0.1,-0.2,-1};
     for(const auto& speed : speeds)
```
Comment (ollyturner) [inline coordinate.cpp:91]: We should use a vector of directions, like line 69, to test all the cases.

File: player.cpp [hunk 4]
```diff
@@ -311,15 +312,17 @@ void test_player() //!OCLINT tests may be long
...
     {
         player p;
-        cordinate predicted_player_position = predict_players_movement(p);
+        coordinate predicted_player_position = predict_players_movement(p);
         p.move();
-        assert(p.get_position()==predicted_player_position);
-        assert(((predicted_player_position.m_x - get_x(p))<0.001)&&(((predicted_player_position.m_x - get_x(p))>-0.001)));
-        assert(((predicted_player_position.m_x - get_y(p))<0.001)&&(((predicted_player_position.m_x - get_y(p))>-0.001)));
+        assert(p.get_position() == predicted_player_position);
+        assert(predicted_player_position.get_x() - get_x(p) < 0.001
```
Comment (ollyturner) [inline player.cpp:322]: We should make an issue to standardise tolerance. We can see various 0s before the 1 throughout the code.

File: game.h [hunk 5]
```diff
@@ -126,7 +126,6 @@ class game
 
   ///Manages collisons with walls
   player resolve_wall_collision(player p);
-
```
Comment (ollyturner) [inline game.h:129]: Why did you remove this line?

File: player.cpp (hunk 4 above)
Comment (ollyturner) [reply player.cpp:322]: We should also standardise the method we use this in.

Comment (ollyturner) [inline player.cpp:322]: To be more consistent, you could check the same way as you did in coordinate.cpp on line 64.

**Review by ollyturner — CHANGES_REQUESTED** [review]
You fixed this problem very efficiently. However, next time you could make it more consistent throughout the PR. Overall, your code was written very well!

File: coordinate.cpp (hunk 2 above)
Comment (TheoPannetier) [reply coordinate.cpp:84]: I have added two more less trivial directions, but I think any more would be overkill, actually.
+ you need to make a separate expectation for every direction, unlike the convenient exemples of the x=0 or y = 0 movement 😅

File: coordinate.cpp (hunk 1 above)
Comment (TheoPannetier) [reply coordinate.cpp:67]: removed

File: game.h (hunk 5 above)
Comment (TheoPannetier) [reply game.h:129]: I don't know. Will revert.

File: player.cpp (hunk 4 above)
Comment (EvoLandEco) [reply player.cpp:322]: To amend @ollyturner 's comment, we should also standardise the method ***of testing whether the error produced is within the tolerance***
Let's make it a separate issue.

Comment (TheoPannetier) [reply player.cpp:322]: I agree, that's a very good idea actually. I have made this an issue (#594 ), do you want to pick it :)

**Review by EvoLandEco — CHANGES_REQUESTED** [review]
Good job! The only thing left is to make sure that you have already reverted the deletion of that line 😸

File: player.cpp (hunk 4 above)
Comment (EvoLandEco) [reply player.cpp]: @TheoPannetier Sure, I will try to implement one 🚀

File: game.h (hunk 5 above)
Comment (TheoPannetier) [reply game.h:129]: Oops, no I haven't! Sorry, shouldn't promise stuff before actually committing the changes 😅

File: game.h [hunk 6]
```diff
@@ -126,7 +126,6 @@ class game
 
   ///Manages collisons with walls
   player resolve_wall_collision(player p);
```
Comment (TheoPannetier) [inline game.h:128]:
```suggestion
  player resolve_wall_collision(player p);
  
```

File: game.h (hunk 5 above)
Comment (TheoPannetier) [reply game.h:129]: done now!
